pages/a/1k1-azure-post.html

```diff
@@ -0,0 +1,239 @@
+    ...
+    "vnetID": "[resourceId(variables('resourceGroupNetworkName'), 'Microsoft.Network/virtualNetworks'
+        , variables('virtualNetworkName'))]",
+    ...
+```
+
+#### Docker Swarm managers
+
+Docker Swarm managers and Consul agents run in the dedicated [Docker Swarm managers group]
+(https://github.com/crate/p-azure/blob/master/arm-templates/docker/swarmmanagers.json).
+
+    * 3 Network Interfaces
+    * 3 Virtual Machine
+    * 6 Microsoft Azure extensions
+    * 1 Availability set
+    * 1 Storage account
+
+#### Crate nodes
+
+All in all, to have 1001 Crate data nodes in the cluster,
+[11 scale unit](https://github.com/crate/p-azure/blob/master/arm-templates/docker/scaleunit.json)
+groups must be deployed. With an idea of keeping the template as simple as possible, while
+satisfying existing constraints, the following structure of the template was proposed:
+
+    * 100 Network Interfaces
+    * 100 Virtual Machine
+    * 200 Microsoft Azure extensions
+    * 1 Availability set
+    * 5 Storage accounts
+
+#### Crate masters
+
+The only difference between the resources in scale and master units is the type of virtual machines
+and their number. Therefore, we use the scale unit
+[ARM template](https://github.com/crate/p-azure/blob/master/arm-templates/docker/scaleunit.json)
+with slightly modified [configuration]
+(https://github.com/crate/p-azure/blob/master/arm-templates/docker masterunit.parameters.json)
+effecting the address space, number of storage account and VM type.
+
+#### Ganglia monitoring
+
+Yet another resource group is
+[Ganglia monitoring](https://github.com/crate/p-azure/tree/master/arm-templates/ganglia). For each
+scale and master unit we assign a dedicated VM to collect and store aggregated metrics into a
+storage engine like [RRD](http://www.fromdual.com/round-robin-database-storage-engine).
+
+#### Deployment
+
+Prepared [ARM templates](https://github.com/crate/p-azure/tree/master/arm-templates) are
+deployed using Azure CLI. Before the template deployment a resource group in a certain location
+must be created.
+
+
+```bash
+azure group create -n <resourcegroup> -l <location>
+azure group deployment create \
+    -f azuredeploy.json \
+    -e azuredeploy.parameters.json \
+    -g <resourcegroup>
+```
+
+
+## Dimensioning
+
+For those interested and keeping count, here's the total Azure VMs we made use of and
+how [Azure references](https://azure.microsoft.com/en-us/documentation/articles/
+virtual-machines-size-specs/#standard-tier-dv2-series) them:
+
+- 12x **Ganglia VMs**: _Standard_D4_v2_ (8 cores, 28GB RAM, 400GB local SSD)
+- 3x **Swarm Manager VMs**: _Standard_D14_v2_ (16 cores, 112GB RAM, 800GB local SSD)
+- 3x **Crate Master VMs**: _Standard_D5_v2_ (16 cores, 56GB RAM, 800GB local SSD)
+- 1100x **Crate Node VMs**: _Standard_D12_v2_ (4 cores, 28GB RAM, 200GB local SSD)
+
+That all adds up to a gigantic **4448** cores and **30968GB** RAM, **just** for the Crate instances.
+
+
```

Comment:
i would state here, what we did. we didn't do much. but maybe one could put this into nice words.